lib/plugins/aws/package/compile/imports/imports.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const traverse = require('traverse');
+
+class AwsCompileImports {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.compileImports = this.compileImports.bind(this);
+
+    this.hooks = {
+      'package:compileImports': this.compileImports,
+    };
+  }
+
+  compileImports() {
+    const that = this; // save this because of the upcoming scope switch
+    traverse(this.serverless.service).forEach(function (propertyParam) {
```

Comment:
gotcha. If we're gonna use the built in variable system via `self` we won't need that I think